File: CS/WebSite/App_Code/MyCallbackCommands.cs
using System;
using System.Data;
using System.Configuration;
using System.Web;
using DevExpress.Web.ASPxScheduler.Internal;
using DevExpress.Web.ASPxScheduler;
using DevExpress.XtraScheduler;
using DevExpress.XtraScheduler.Commands;

public class MySwitchViewCallbackCommand : SwitchViewCallbackCommand {
    public MySwitchViewCallbackCommand(ASPxScheduler control)
        : base(control) {
    }
    protected override void ExecuteCore() {
        base.ExecuteCore();
        if (Control.ActiveViewType == SchedulerViewType.Month)
            Control.Start = DateTimeNavigationHelper.CalculateMonthFirstDate(Control.Start);
    }
}
public class MyNavigateForwardCallbackCommand : NavigateForwardCallbackCommand {
    public MyNavigateForwardCallbackCommand(ASPxScheduler control)
        : base(control) {
    }
    protected override SchedulerCommand CreateCommand() {
        return new MyNavigateViewForwardCommand(Control);
    }
}
public class MyNavigateBackwardCallbackCommand : NavigateBackwardCallbackCommand {
    public MyNavigateBackwardCallbackCommand(ASPxScheduler control)
        : base(control) {
    }
    protected override SchedulerCommand CreateCommand() {
        return new MyNavigateViewBackwardCommand(Control);
    }
}
public class MyNavigateViewForwardCommand : NavigateViewForwardCommand {
    public MyNavigateViewForwardCommand(ISchedulerCommandTarget target)
        : base(target) {
    }
    protected override void ExecuteCore() {
        InnerControl.Start = DateTimeNavigationHelper.CalculateNextMonthStartDate(InnerControl.Start);
    }
}
public class MyNavigateViewBackwardCommand : NavigateViewBackwardCommand {
    public MyNavigateViewBackwardCommand(ISchedulerCommandTarget target)
        : base(target) {
    }
    protected override void ExecuteCore() {
        InnerControl.Start = DateTimeNavigationHelper.CalculatePrevMonthStartDate(InnerControl.Start);
    }
}
public static class DateTimeNavigationHelper {
    public static DateTime CalculateNextMonthStartDate(DateTime startDate) {
        DateTime newDate = startDate.AddMonths(1);
        if (newDate.Day > 22)
            newDate = newDate.AddMonths(1);
        newDate = CalculateMonthFirstDate(newDate);
        return newDate;
    }
    public static DateTime CalculatePrevMonthStartDate(DateTime startDate) {
        DateTime newDate = startDate.AddMonths(-1);
        if (newDate.Day > 22)
            newDate = newDate.AddMonths(1);
        newDate = CalculateMonthFirstDate(newDate);
        return newDate;
    }
    public static DateTime CalculateMonthFirstDate(DateTime newDate) {
        DateTime result;
        result = newDate.Date.AddDays(1 - newDate.Day);
        return result;
    }
}

File: VB/WebSite/App_Code/MyCallbackCommands.vb
Imports Microsoft.VisualBasic
Imports System
Imports System.Data
Imports System.Configuration
Imports System.Web
Imports DevExpress.Web.ASPxScheduler.Internal
Imports DevExpress.Web.ASPxScheduler
Imports DevExpress.XtraScheduler


Public Class MyNavigateForwardCallbackCommand
	Inherits NavigateForwardCallbackCommand
	Public Sub New(ByVal control As ASPxScheduler)
		MyBase.New(control)
	End Sub

	Protected Overrides Sub ExecuteCore()
		Dim plusMonthDate As DateTime = Control.Start.Date.AddMonths(1)
		If plusMonthDate.Day > 22 Then
			plusMonthDate = plusMonthDate.Date.AddMonths(1).AddDays(1 - plusMonthDate.Day)
		Else
			plusMonthDate = plusMonthDate.Date.AddDays(1 - plusMonthDate.Day)
		End If
		MyBase.ExecuteCore()

		If Control.ActiveViewType = SchedulerViewType.Month Then
			Control.Start = plusMonthDate
		End If

	End Sub
End Class